### ISSUE TITLE
Adding Sprite Mask support natively instead of using shader workarounds.

### ISSUE THREAD

**Issue by boraty14** [defold/defold#8494]
Defold is a great engine, especially for making 2d games. I know that Defold's general philosophy is that it gives you the fundamental building blocks, and you add them as you need it. But I believe having Sprite Mask's are needed quite often enough to consider out of the box support for it. (Slot games, creating splash effects on ground/walls with mask/ making a detective games with hidden items etc... examples vary I believe). 

I would personally would like to be able to set it from editor via url, also set it dynamically in code in runtime.

https://forum.defold.com/t/how-to-mask-clipping-a-sprite/28706
https://forum.defold.com/t/another-sprite-masking-question/69203
https://forum.defold.com/t/solved-how-to-make-sprite-only-visible-in-certain-rectangle-area/70925
https://github.com/jhonnyking/my-public-defold-examples/tree/master/sprite-masking

Various questions about it and also I could find two custom solutions for it, one is altering with the stencil buffer via rendering scripts and other just hardcodes the sprite area to shader. I believe having a more unified solution would be nicer. I think while both solutions are nice and can save the day, having this by default would make more sense.

It does not have to be built in exactly to the editor I believe. It would be more than enough if it was a defold asset/extension with some documentation and design for general use.

**Comment by britzl** [comment]
Related: https://github.com/defold/defold/issues/8609

**Comment by NANAMINER** [comment]
You are absolutely right! I was very surprised that Defold does not support the sprite mask when I switched to it. This is such a basic and necessary function!
